### PR TITLE
Added migration support for CMake CUDA library variables and targets

### PR DESCRIPTION
**Pull request by TejaX-Alaghari**
Added migration support for following CMake CUDA library variables and targets: 

- CUDA_CUFFT_LIBRARIES {MKL}
- CUDA_CUBLAS_LIBRARIES {MKL}
- CUDA_curand_LIBRARY {MKL}
- CUDA_cusolver_LIBRARY {MKL}
- CUDA_cusparse_LIBRARY {MKL}
----
- CUDA_LIBRARIES {NA}
- CUDA_cudart_static_LIBRARY {NA}
- CUDA_cudadevrt_LIBRARY {NA}
- CUDA_cupti_LIBRARY {NA}
- CUDA_CUT_LIBRARY {NA}
- CUDA_npp_LIBRARY {NA}
- CUDA_nppc_LIBRARY {NA}
- CUDA_nppi_LIBRARY {NA}
- CUDA_nppial_LIBRARY {NA}
- CUDA_nppicc_LIBRARY {NA}
- CUDA_nppicom_LIBRARY {NA}
- CUDA_nppidei_LIBRARY {NA}
- CUDA_nppif_LIBRARY {NA}
- CUDA_nppig_LIBRARY {NA}
- CUDA_nppim_LIBRARY {NA}
- CUDA_nppist_LIBRARY {NA}
- CUDA_nppisu_LIBRARY {NA}
- CUDA_nppitc_LIBRARY {NA}
- CUDA_npps_LIBRARY {NA}
- CUDA_nvcuvenc_LIBRARY {NA}
- CUDA_nvcuvid_LIBRARY {NA}
- CUDA_nvToolsExt_LIBRARY {NA}
----
- CUDA::cufft {MKL}
- CUDA::cufftw {MKL}
- CUDA::cufft_static {MKL}
- CUDA::cufft_static_nocallback {MKL}
- CUDA::cufftw_static {MKL}
- CUDA::curand {MKL}
- CUDA::curand_static {MKL}
- CUDA::cusolver {MKL}
- CUDA::cusolver_static {MKL}
- CUDA::cusparse {MKL}
- CUDA::cusparse_static {MKL}
----
- CUDA::cudla {NA}
- CUDA::cuFile {NA}
- CUDA::cuFile_static {NA}
- CUDA::cuFile_rdma {NA}
- CUDA::cuFile_rdma_static {NA}
- CUDA::cupti {NA}
- CUDA::cupti_static {NA}
- CUDA::nvperf_host {NA}
- CUDA::nvperf_host_static {NA}
- CUDA::nvperf_target {NA}
- CUDA::pcsamplingutil {NA}
- CUDA::nppc {NA}
- CUDA::nppc_static {NA}
- CUDA::nppial {NA}
- CUDA::nppial_static {NA}
- CUDA::nppicc {NA}
- CUDA::nppicc_static {NA}
- CUDA::nppicom {NA}
- CUDA::nppicom_static {NA}
- CUDA::nppidei {NA}
- CUDA::nppidei_static {NA}
- CUDA::nppif {NA}
- CUDA::nppif_static {NA}
- CUDA::nppig {NA}
- CUDA::nppig_static {NA}
- CUDA::nppim {NA}
- CUDA::nppim_static {NA}
- CUDA::nppist {NA}
- CUDA::nppist_static {NA}
- CUDA::nppisu {NA}
- CUDA::nppisu_static {NA}
- CUDA::nppitc {NA}
- CUDA::nppitc_static {NA}
- CUDA::npps {NA}
- CUDA::npps_static {NA}
- CUDA::nvblas {NA}
- CUDA::nvgraph {NA}
- CUDA::nvgraph_static {NA}
- CUDA::nvjpeg {NA}
- CUDA::nvjpeg_static {NA}
- CUDA::nvptxcompiler_static {NA}
- CUDA::nvrtc_builtins {NA}
- CUDA::nvrtc_static starting {NA}
- CUDA::nvrtc_builtins_static {NA}
- CUDA::nvJitLink starting {NA}
- CUDA::nvJitLink_static {NA}
- CUDA::nvml {NA}
- CUDA::nvtx3 {NA}
- CUDA::OpenCL {NA}
----

Note: SYCL target is displayed inside {} against each CMake CUDA syntax

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by zhiweij1 — APPROVED** [review]

File: clang/test/dpct/cmake_migration/case_032/expected.txt [hunk 1]
```diff
@@ -29,3 +29,108 @@ else ()
     set(LLAMA_EXTRA_LIBS ${LLAMA_EXTRA_LIBS}  ${MKL_LIB} ${DNN_LIB})
 endif()
 set(LLAMA_EXTRA_LIBS ${LLAMA_EXTRA_LIBS}  ${MKL_LIB} ${DNN_LIB})
+
+set(LIBS ${LIBS}
+
```
Comment (zhimingwang36) [inline clang/test/dpct/cmake_migration/case_032/expected.txt:34]: Please try to further process the empty lines by removing them.  May need an update on infrastructure.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

File: clang/test/dpct/cmake_migration/case_032/expected.txt (hunk 1 above)
Comment (tomflinda) [reply clang/test/dpct/cmake_migration/case_032/expected.txt:34]: Agree, I will refine infrastructure to enhance it.

**Pull request merged**